app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,25 @@
+class InvitationsController < ApplicationController
+  before_action :require_user
+
+  def new
+    @invitation = Invitation.new
+  end
+
+  def create
+    @invitation = Invitation.new(invitation_params.merge!(inviter_id: current_user.id))
+    if @invitation.save
+      ApplicationMailer.send_invitation_email(@invitation).deliver
+      flash[:success] = "Your invitation has been sent!"
+      redirect_to new_invitation_path
+    else
+      flash[:danger] = "All fields must be completed."
```

Comment:
Whenever you render you need to use `flash.now`. If you use `flash`, the flash message will survive the next redirect/render as well.